chore: devcontainer with codespaces

atian25 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * GitHub Codespaces and VS Code Dev Containers support for ready-to-code development environment
  * Pre-configured services: MySQL and Redis, with PostgreSQL support available
  * Automated database initialization and dependency installation on environment startup

* **Documentation**
  * Added comprehensive Dev Container configuration guide and developer quick-start instructions

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

fengmk2 commented on test/TestUtil.ts:
```diff
@@ -42,10 +42,10 @@ export class TestUtil {
 
   static getMySqlConfig() {
     return {
-      host: process.env.MYSQL_HOST || 'localhost',
```
这个修改了本地起 mysql 开发就不行了。

thonatos commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,30 @@
+	"forwardPorts": [
+		"phpmyadmin:80"
+	],
+
```
```json
"containerEnv": {
  "MYSQL_HOST": "mysql"
},
```

这样应该就可以了，cc @atian25

fengmk2 commented:
@thonatos 还继续么？

fengmk2 committed:
feat: devcontainer with codespaces

Add a self-contained .devcontainer setup so cnpmcore can be developed in
GitHub Codespaces or the VS Code Dev Containers extension with no manual setup:

- docker-compose stack: Node.js 22 workspace + mysql:9 + redis:6-alpine
- Dockerfile installs the mysql/postgresql client CLIs that the
  prepare-database scripts shell out to
- post-create.sh installs deps and initializes the cnpmcore dev database
  (MySQL by default, PostgreSQL switchable)
- forward registry port 7001, recommend oxc/docker/actions extensions
- README/DEVELOPER docs plus an "Open in GitHub Codespaces" badge

fengmk2 commented:
<img width="3448" height="2084" alt="image" src="https://github.com/user-attachments/assets/ac9c3bad-493a-4a9d-ae90-eb749bebcefd" />
<img width="1302" height="1056" alt="image" src="https://github.com/user-attachments/assets/be850e65-125e-43cb-925c-041dcdffc6af" />

works
